lib/AST/ASTMangler.cpp

```diff
@@ -3904,11 +3913,27 @@ ASTMangler::mangleOpaqueTypeDescriptorRecord(const OpaqueTypeDecl *decl) {
   return finalize();
 }
 
+std::string ASTMangler::mangleDistributedThunkRef(const AbstractFunctionDecl *thunk) {
+  bool shouldEnableThunkDeclMangling = false;
+  if (auto extension = dyn_cast<ExtensionDecl>(thunk->getDeclContext())) {
+    if (auto extended = extension->getExtendedNominal()) {
+      shouldEnableThunkDeclMangling = isa<ProtocolDecl>(extended);
+    }
+  }
+  // Marker protocols cannot be checked at runtime, so there is no point
+  // in recording them for distributed thunks.
+  llvm::SaveAndRestore<bool> savedDistributedMangleProtocolTargetAsStubTarget(
+      DistributedMangleProtocolTargetAsStubTarget, shouldEnableThunkDeclMangling);
```

Comment:
That's nicer, thank you :) Working on it

Reply:
I did the entry points but sadly we do need the booleans -- otherwise we end up $ prefixing more than we intend to -- other types included in the mangleEntity would wrongly get prefixed then as well.

I think this is a lesser evil than doing a whole "copy the entire mangling body into the mangle... function" here.

Reply:
I'll see if I could reduce this to one boolean though -- I think that would work